[Bug] README/schema sync already broken

### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The bug has not been fixed in the latest version.
- [x] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [x] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/SpecForge/discussions/new/choose Otherwise, it will be closed.
- [x] 5. Please use English, otherwise it will be closed.

### Describe the bug

tests/test_config/test_recipe_readme.py is meant to keep examples/configs/README.md in sync with the typed config schema, but the test never actually runs in CI.

It is written as a bare module-level function (def test_recipe_readme_names_every_typed_config_field():, no unittest.TestCase), while the CI unit-test job runs python -m unittest discover, which only collects unittest.TestCase subclasses. The module is imported but 0 cases are collected, so the guard has been dead code since it was introduced.

Drift has already slipped through: #665 added use_liger_kernel to ModelConfig in specforge/config/schema.py without the corresponding row in examples/configs/README.md, and its CI was legitimately green because the guard never executes.

### Reproduction

No model or GPU required; test-infrastructure repro on a plain checkout.

1) What CI does — silent no-op:
$ python -m unittest tests.test_config.test_recipe_readme -v
Ran 0 tests in 0.000s
NO TESTS RAN

2) What the guard would say if it actually ran:
$ python -m pytest tests/test_config/test_recipe_readme.py -q
FAILED tests/test_config/test_recipe_readme.py::test_recipe_readme_names_every_typed_config_field
AssertionError: examples/configs/README.md is missing config fields: ['model.use_liger_kernel']

3) Confirm the drift:
$ grep -c "use_liger_kernel" specforge/config/schema.py   # -> 1 (ModelConfig)
$ grep -c "use_liger_kernel" examples/configs/README.md   # -> 0

CI command reference: .github/workflows/test.yaml, "Run test" step:
python -m unittest discover -s ./tests -p "test_*.py" -v

### Environment

- Repo: sgl-project/SpecForge @ main (reproduced on b73aa89 and on current main fec8f85)
- Python: 3.11 (same as CI), no torch/GPU needed — specforge.config imports fine without them
- CI: .github/workflows/test.yaml, unit-test job, container lmsysorg/sglang:v0.5.14-cu130

### Suggested fix:
1. Wrap the check in a TestCase (e.g. class TestRecipeReadmeSync(unittest.TestCase)) so unittest discover picks it up — keeps the existing unittest CI untouched.
2. In the SAME PR, add the missing `model.use_liger_kernel` row to examples/configs/README.md, otherwise the newly-live guard goes red on main.
(Alternative: switch the CI step to pytest, which collects bare functions — but that changes collection semantics for the entire suite.)

--- a/examples/configs/README.md
+++ b/examples/configs/README.md
@@ -146,6 +146,7 @@ should make their training strategy and topology explicit.
 | `model.input_modality` | `text` | The provider modality. The unified runtime supports text only; VLM modalities such as `qwen2_5_vl` are rejected. |
 | `model.shard_target_output` | `false` | Retained for config migration; leave it false on the server-only online path. |
 | `model.trust_remote_code` | `false` | Enable only for model repositories that require custom loading code. |
+| `model.use_liger_kernel` | `false` | Enable Liger Qwen3 RMSNorm/SwiGLU kernels for DFlash training. Requires the `specforge[liger]` extra. |
 | `model.embedding_key` | `model.embed_tokens.weight` | Target checkpoint key copied into or used by the draft embedding. |
 | `model.lm_head_key` | `lm_head.weight` | Target checkpoint key used for the frozen output head. |
 | `model.vocab_mapping_path` | `""` | Target-to-draft vocabulary mapping. EAGLE3 disaggregated runs require an explicit shared file. |

--- a/tests/test_config/test_recipe_readme.py
+++ b/tests/test_config/test_recipe_readme.py
@@ -1,5 +1,6 @@
 """Keep the user-facing YAML reference synchronized with the typed schema."""
 
+import unittest
 from pathlib import Path
 
 from specforge.config import (
@@ -22,37 +23,40 @@ ROOT = Path(__file__).resolve().parents[2]
 README = ROOT / "examples" / "configs" / "README.md"
 
 
-def test_recipe_readme_names_every_typed_config_field():
-    text = README.read_text(encoding="utf-8")
-    schemas = (
-        ("", Config),
-        ("model", ModelConfig),
-        ("data", DataConfig),
-        ("training", TrainingConfig),
-        ("tracking", TrackingConfig),
-        ("profiling", ProfilingConfig),
-        ("runtime", RuntimeConfig),
-        ("deployment", DeploymentConfig),
-        ("deployment.trainer", TrainerDeploymentConfig),
-        ("deployment.disaggregated", DisaggregatedDeploymentConfig),
-        ("deployment.disaggregated.managed_local", ManagedLocalStackConfig),
-        (
-            "deployment.disaggregated.managed_local.mooncake",
-            ManagedLocalMooncakeConfig,
-        ),
-        (
-            "deployment.disaggregated.managed_local.capture_servers[]",
-            ManagedLocalCaptureServerConfig,
-        ),
-    )
+class TestRecipeReadmeSync(unittest.TestCase):
+    def test_recipe_readme_names_every_typed_config_field(self):
+        text = README.read_text(encoding="utf-8")
+        schemas = (
+            ("", Config),
+            ("model", ModelConfig),
+            ("data", DataConfig),
+            ("training", TrainingConfig),
+            ("tracking", TrackingConfig),
+            ("profiling", ProfilingConfig),
+            ("runtime", RuntimeConfig),
+            ("deployment", DeploymentConfig),
+            ("deployment.trainer", TrainerDeploymentConfig),
+            ("deployment.disaggregated", DisaggregatedDeploymentConfig),
+            ("deployment.disaggregated.managed_local", ManagedLocalStackConfig),
+            (
+                "deployment.disaggregated.managed_local.mooncake",
+                ManagedLocalMooncakeConfig,
+            ),
+            (
+                "deployment.disaggregated.managed_local.capture_servers[]",
+                ManagedLocalCaptureServerConfig,
+            ),
+        )
 
-    missing = []
-    for prefix, schema in schemas:
-        for field_name in schema.model_fields:
-            path = f"{prefix}.{field_name}" if prefix else field_name
-            if f"`{path}`" not in text:
-                missing.append(path)
+        missing = []
+        for prefix, schema in schemas:
+            for field_name in schema.model_fields:
+                path = f"{prefix}.{field_name}" if prefix else field_name
+                if f"`{path}`" not in text:
+                    missing.append(path)
 
-    assert (
-        not missing
-    ), f"examples/configs/README.md is missing config fields: {missing}"
+        self.assertEqual(
+            [],
+            missing,
+            f"examples/configs/README.md is missing config fields: {missing}",
+        )

